src/functionality.js

```diff
@@ -0,0 +1,162 @@
+    checkbox.type = 'checkbox';
+    checkbox.addEventListener('click', (event) => {
+      const { index } = event.target.parentNode.dataset;
+      tasks[index].completed = event.target.checked;
+      saveTasksToLocalStorage();
+
+      if (event.target.checked) {
+        event.target.parentNode.classList.add('completed');
+      } else {
+        event.target.parentNode.classList.remove('completed');
+      }
+    });
+
+    const taskName = document.createTextNode(`   ${newTask.description}`);
+    listItem.appendChild(checkbox);
+    listItem.appendChild(document.createTextNode('   '));
+    listItem.appendChild(taskName);
+
+    const moreButton = document.createElement('button');
+    const editButton = document.createElement('button');
+    editButton.textContent = 'Edit';
+    editButton.classList.add('edit-button');
+    editButton.addEventListener('click', () => {
+      const inputField = document.createElement('input');
+      inputField.type = 'text';
+      inputField.value = taskName.nodeValue.trim();
+      inputField.addEventListener('keydown', (e) => {
+        if (e.key === 'Enter') {
+          taskName.nodeValue = `   ${inputField.value.trim()}`;
+          listItem.removeChild(inputField);
+          listItem.appendChild(checkbox);
+          listItem.appendChild(document.createTextNode('   '));
+          listItem.appendChild(taskName);
+          editButton.style.display = 'none';
+          moreButton.style.display = 'inline-block';
+          tasks[listItem.dataset.index].description = inputField.value.trim();
+          saveTasksToLocalStorage();
+        }
+      });
+
+      listItem.removeChild(taskName);
+      listItem.appendChild(inputField);
+      inputField.focus();
+    });
+
+    moreButton.textContent = '•••';
+    moreButton.classList.add('more-button');
+    moreButton.addEventListener('click', () => {
+      editButton.style.display = 'inline-block';
+      moreButton.style.display = 'none';
+    });
+
+    moreButton.style.float = 'right';
+    moreButton.style.marginRight = '10px';
+    moreButton.style.border = 'none';
+    moreButton.style.backgroundColor = 'white';
+    editButton.style.float = 'right';
+    editButton.style.marginRight = '10px';
+
+    listItem.appendChild(moreButton);
+    listItem.appendChild(editButton);
+    editButton.style.display = 'none';
+
+    listItem.dataset.index = tasks.length - 1;
+    todoList.appendChild(listItem);
+
+    const hr = document.createElement('hr');
+    todoList.appendChild(hr);
+
+    input.value = '';
+    errorMessage.textContent = '';
+  } else {
+    errorMessage.textContent = 'Task cannot be empty';
+  }
+};
+
+const input = document.getElementById('type-todo');
+input.addEventListener('keydown', (event) => {
+  if (event.key === 'Enter') {
+    event.preventDefault();
+    addTask();
+  }
+});
+
+function deleteTask(index) {
+  tasks.splice(index, 1);
+  updateIndexes();
+  saveTasksToLocalStorage();
+  renderTodoList();
+}
```

Comment:
- [OPTIONAL] I am only requiring this as optional because this requirement was missed by the previous reviewer. In your app I dont see a deleting button or icon for individual todo. the only way to delete a todo is by checking it completed and then pressing clear all button. kindly consider implementing this deleteTask method on individual todo on a deleting icon in every todo task.
design guidelines
![qw](https://github.com/uno36/To-do-list/assets/96683843/2960c70b-9091-4506-ad7d-337f2f10da5d)